### ISSUE TITLE
Output YAML

### ISSUE THREAD

**Issue by probonopd** [azubieta/appimage-releases-indexer#4]
Would it be possible to output YAML instead of and/or in addition to JSON?

**Comment by azubieta** [comment]
Possible it is. But I don't know if it would be convenient. This output is meant to be consumed by other apps not by people. So Json was chosen for being more machine friendly.

Would you mind to explain a bit the use cases you have in mind.

**Comment by azubieta** [comment]
By the way we should move the "AppInfo" spec to another place were we could discuss it.

**Comment by probonopd** [comment]
Currently I am using a [crude script](https://github.com/AppImage/appimage.github.io/blob/master/code/worker.sh) to produce https://github.com/AppImage/appimage.github.io/blob/master/apps/ and I was wondering if I could use appimage-releases-indexer instead.

Example:
https://raw.githubusercontent.com/AppImage/appimage.github.io/master/apps/Addaps.md

If basically collects data from
* desktop file
* AppStream XML file
* electron JSON file
and combines it into a single YAML/md file needed by [Jekyll](https://jekyllrb.com/).

**Comment by probonopd** [comment]
> the "AppInfo" spec

Which spec do you mean?

**Comment by azubieta** [comment]
I mean [this](https://github.com/azubieta/appimage-releases-indexer/blob/master/output-example.json) currently is just the output but I was thinking that it would be useful that every AppImage to include it's own. It could even be generated by a linuxdeploy-plugin at the end of the AppImage construction process. Also could include the functionality to push it (publish) to our listings server.

**Comment by probonopd** [comment]
Why should we do that as long as it can be constructed from the desktop file and the AppStream file that many apps already have?

**Comment by azubieta** [comment]
About the real topic of this issue:
What about using something like [this](https://www.json2yaml.com/) to properly transform the output to YAML?

**Comment by probonopd** [comment]
> If the data contained in the AppStream format is enough for our purposes of keeping a nice listing of the AppImage available out there then we must use it.

I agree that at the outside AppStream appears to be complicated, especially because it keeps changing (hence you are working against a moving target) and is XML (which is just a pain to deal with by hand). But apart from that, it has similar to objectives to what you are trying to build from scratch here, so maybe you could exchange some ideas with the AppStream guys.

cc @ximion

**Comment by ximion** [comment]
@probonopd AppStream has a YAML variant for it's collection metadata flavor (so, it can't be used by applications, but by people making software stores).
You can run `appstreamcli convert <xml-file-path>` to get YAML.

As for AppStream being a moving target: The specification hasn't changed for a long time, the last breaking change was in 2016 and only affected the collection metadata. However, the validator gets much better with every release at catching mistakes automatically.

As for adding features to AppStream: Filing an [issue](https://github.com/ximion/appstream/issues) about the desired feature is usually the best idea, coming with a patch is even better. However, adding new stuff usually comes with an extensive discussion and might take a while (unless it's completely uncontroversial).
